plugins/modules/sda_host_port_onboarding_workflow_manager.py

```diff
@@ -0,0 +1,3010 @@
+        Returns:
+            dict: A dictionary containing the parameters 'fabric_id' and 'network_device_id'.
+        Description:
+            This method creates a dictionary with 'fabric_id' and 'network_device_id' parameters required
+            for retrieving port assignments. It logs the generated parameters for debugging purposes and
+            returns the dictionary.
+        """
+        # Create dictionary with required parameters
+        get_port_assignment_params = {
+            "fabric_id": fabric_id,
+            "network_device_id": network_device_id,
+        }
+
+        self.log("get_port_assignments_params: {0}".format(get_port_assignment_params), "DEBUG")
+
+        return get_port_assignment_params
+
+    def get_port_assignments(self, get_port_assignments_params):
+        """
+        Retrieves port assignments from Cisco Catalyst Center using the given parameters.
+        Args:
+            get_port_assignments_params (dict): Parameters for querying port assignments, including fabric ID and network device ID.
+        Returns:
+            list: A list of port assignments retrieved from the API.
+        Description:
+            This method retrieves port assignments from Cisco Catalyst Center by executing the 'get_port_assignments' API call.
+            It uses pagination with offset and limit to handle large datasets. The method logs relevant information and returns
+            the list of port assignments. If an error occurs during the API call, it logs an error message and sets the validation
+            status to "failed".
+        """
+        try:
+            offset = 1
+            limit = 500
+            port_assignments = []
+
+            while True:
+                # Update offset and limit in the parameters
+                get_port_assignments_params.update({
+                    "offset": offset,
+                    "limit": limit
+                })
+
+                self.log("Updated get_port_assignments_params with offset and limit: {0} ".format(get_port_assignments_params), "DEBUG")
+
+                # Execute the API call to get extranet policie
+                response = self.dnac._exec(
+                    family="sda",
+                    function="get_port_assignments",
+                    op_modifies=True,
+                    params=get_port_assignments_params,
+                )
+
+                # Process the response if available
+                response = response.get("response")
+                if not response:
+                    self.log(
+                        "Exiting the loop because no port assignments were returned after increasing the offset. "
+                        "Current offset: {0}".format(offset),
+                        "INFO"
+                    )
+                    break
+
+                port_assignments.extend(response)
+                offset += limit
+                return port_assignments
```

Comment:
if you return inside while, then it will execute only one time.. 
Can you check and move this outside while? 

`        return port_assignments 
`